### PR TITLE
fix xml comments

### PR DESCRIPTION
**Pull request by jennyf19**
add overrides with delegates
#84 

@pmaytak after updating samples, decided to address the suggestion you had earlier this week and just do it now. makes the api much simpler.

### CODE REVIEW

File: src/Microsoft.Identity.Web/WebAppServiceCollectionExtensions.cs [hunk 1]
```diff
@@ -40,20 +47,60 @@ public static class WebAppServiceCollectionExtensions
...
+            bool subscribeToOpenIdConnectMiddlewareDiagnosticsEvents = false)
+        {
+            AuthenticationBuilder builder = services.AddAuthentication(openIdConnectScheme);
+            builder.AddSignIn(
+                 configureOpenIdConnectOptions,
+                 configureMicrosoftIdentityOptions,
+                 openIdConnectScheme,
+                 cookieScheme,
+                 subscribeToOpenIdConnectMiddlewareDiagnosticsEvents);
+            return services;
```
Comment (pmaytak) [inline src/Microsoft.Identity.Web/WebAppServiceCollectionExtensions.cs:90]: I think we need to return `builder` object here, which actually has the `Services` property in itself. From what I've seen in the ASP.NET Core `IServiceCollection` extensions classes, if the extension method only adds some services to the collection, then it's fine to return `services`, but if there's more setup using a builder, then builder is returned. Also `AuthenticaitonBuilder` has some specific auth related methods that can be chained. For example:
```CSharp
services.AddAuthentication(X).AddSignIn(Y).AddCookie(Z).Services.AddLogging();
```

Comment (jennyf19) [reply src/Microsoft.Identity.Web/WebAppServiceCollectionExtensions.cs:90]: like the idea. let's dicuss

Comment (jmprieur) [reply src/Microsoft.Identity.Web/WebAppServiceCollectionExtensions.cs:90]: I get your point, @pmaytak. Thanks for this feedback.

Today, by returning services, we enable (simple) code like:

```csharp
services.AddSignIn(Configuration)
             .AddProtectedWebApiCallsWebAPi(Configuration)
             .AddTokenCacheXXX();
```

If we returned a builder, the code would have to be

```csharp
services.AddSignIn(Configuration);
services.AddProtectedWebApiCallsWebAPi(Configuration)
             .AddTokenCacheXXX();
```

which is ok, but I like a bit less, and given we are already taking care of cookies ...

Finally, it's always possible for customers who want to get a builder to use the following override:

```csharp
services.AddAuthentication(OpenIdConnectAuthentication.Default)
             .AddSignIn(Configuration)
             .AddCookie(Y);

services.AddProtectedWebApiCallsWebAPi(Configuration)
             .AddTokenCacheXXX();
```

Comment (pmaytak) [reply src/Microsoft.Identity.Web/WebAppServiceCollectionExtensions.cs:90]: If we returned a builder, your second code example could be this:
```CSharp
services.AddSignIn(Configuration);
        .Services
        .AddProtectedWebApiCallsWebAPi(Configuration)
        .AddTokenCacheXXX();
```
Similar to code in [ServiceCollectionTests.cs](https://github.com/dotnet/aspnetcore/blob/43c2c9ea97dfacce87f6bd578c1863ee0bf677c4/src/DataProtection/DataProtection/test/ServiceCollectionTests.cs#L15)

I agree with your point of making the API simple, and both of the ways are pretty close in that regard. The builder does provide more authentication setup functionality while still providing access to the `Services` property for chaining. I guess my perspective is a bit more conceptual - do we return a more specific `AuthenticationBuilder` or a more general `IServicesCollection`.

Comment (jmprieur) [reply src/Microsoft.Identity.Web/WebAppServiceCollectionExtensions.cs:90]: Proposing to do a user study on that
(ask users what they prefer)?

**Pull request merged**